pkg/virtctl/create/clone/clone.go

```diff
@@ -0,0 +1,288 @@
+  
+  # Create a manifest for a clone with label filters:
+  {{ProgramName}} create clone --source-name sourceVM --label-filter "*" --label-filter "!some/key" 
+  
+  # Create a manifest for a clone with annotation filters:
+  {{ProgramName}} create clone --source-name sourceVM --annotation-filter "*" --annotation-filter "!some/key"
+
+  # Create a manifest for a clone with new MAC addresses:
+  {{ProgramName}} create clone --source-name sourceVM --new-mac-address interface1:00-11-22 --new-mac-address interface2:00-11-33
+
+  # Create a manifest for a clone with new SMBIOS serial:
+  {{ProgramName}} create clone --source-name sourceVM --new-smbios-serial "new-serial"
+
+  # Create a manifest for a clone and use it to create a resource with kubectl
+  {{ProgramName}} create clone --source-name sourceVM | kubectl create -f -`
+}
+
+func (c *createClone) newClone() (*clonev1alpha1.VirtualMachineClone, error) {
+	clone := kubecli.NewMinimalClone(c.name)
+
+	source, err := c.typeToTypedLocalObjectReference(c.sourceType, c.sourceName, true)
+	if err != nil {
+		return nil, err
+	}
+
+	target, err := c.typeToTypedLocalObjectReference(c.targetType, c.targetName, false)
+	if err != nil {
+		return nil, err
+	}
+
+	clone.Spec = clonev1alpha1.VirtualMachineCloneSpec{
+		Source:            source,
+		Target:            target,
+		AnnotationFilters: c.annotationFilters,
+		LabelFilters:      c.labelFilters,
+		NewSMBiosSerial:   pointer.P(c.newSmbiosSerial),
+	}
+
+	return clone, nil
+}
+
+func (c *createClone) applyFlags(cmd *cobra.Command, spec *clonev1alpha1.VirtualMachineCloneSpec) error {
+	for flag := range optFns {
+		if cmd.Flags().Changed(flag) {
+			if err := optFns[flag](c, (*cloneSpec)(spec)); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *createClone) run(cmd *cobra.Command) error {
+	c.setDefaults()
+	err := c.validateFlags()
+	if err != nil {
+		return err
+	}
+
+	clone, err := c.newClone()
+	if err != nil {
+		return err
+	}
+
+	err = c.applyFlags(cmd, &clone.Spec)
+	if err != nil {
+		return err
+	}
+
+	if clone.Name == "" {
+		clone.Name = "clone-" + rand.String(5)
+	}
+
+	cloneBytes, err := yaml.Marshal(clone)
+	if err != nil {
+		return err
+	}
+
+	cmd.Print(string(cloneBytes))
+	return nil
+}
+
+func (c *createClone) typeToTypedLocalObjectReference(sourceOrTargetType, sourceOrTargetName string, isSource bool) (*v1.TypedLocalObjectReference, error) {
+	var kind, apiGroup string
+
+	generateErr := func() error {
+		var sourceTargetStr string
+		var supportedTypes string
+
+		if isSource {
+			sourceTargetStr = "source"
+			supportedTypes = supportedSourceTypes
+		} else {
+			sourceTargetStr = "target"
+			supportedTypes = supportedTargetTypes
+		}
+
+		return fmt.Errorf("%s type %s is not supported. Supported types: %s", sourceTargetStr, sourceOrTargetType, supportedTypes)
+	}
+
+	switch sourceOrTargetType {
+	case "vm", "VM", "VirtualMachine", "virtualmachine":
```

Comment:
nit - Could we document these in the help output somehow? You're only listing `vm` and `vmsnapshot` at the moment.